server/src/test/java/org/elasticsearch/action/bulk/BulkProcessorRetryIT.java

```diff
@@ -114,9 +114,9 @@ public void afterBulk(long executionId, BulkRequest request, Throwable failure)
                 for (BulkItemResponse bulkItemResponse : bulkResponse.getItems()) {
                     if (bulkItemResponse.isFailed()) {
                         BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
-                        Throwable rootCause = ExceptionsHelper.unwrapCause(failure.getCause());
-                        if (rootCause instanceof EsRejectedExecutionException) {
-                            if (rejectedExecutionExpected == false) {
+                        if (failure.getStatus() == RestStatus.TOO_MANY_REQUESTS) {
+                            if (!rejectedExecutionExpected) {
+                                Throwable rootCause = ExceptionsHelper.unwrapCause(failure.getCause());
```

Comment:
maybe we do not even need to unwrap it here anymore? could we just use getCause instead when throwing assertion error below?

Reply:
I see that this has not been addressed. Is that on purpose? I see that you have done this on the client version of the test so it should be fine here too.